objc/objc_runtime_darwin.go

```diff
@@ -148,203 +167,154 @@ func AllocateClassPair(super Class, name string, extraBytes uintptr) Class {
 		}
-		if !class.AddMethod(sel, imp, encoding) {
-			return 0, fmt.Errorf("objc: couldn't add Method %s", met.Name)
+		if !class.AddMethod(def.Cmd, imp, encoding) {
+			return 0, fmt.Errorf("objc: couldn't add Method at index %d", idx)
 		}
 	}
 	// Add Ivars
 	// Start at 1 because we skip the class object which is first
-	for i := 1; i < strct.NumField(); i++ {
-		f := strct.Field(i)
-		if f.Name == "_" {
-			continue
-		}
-		size := f.Type.Size()
-		alignment := uint8(math.Log2(float64(f.Type.Align())))
-		enc, err := encodeType(f.Type, false)
+	for _, ivar := range ivars {
+		size := ivar.Type.Size()
+		alignment := uint8(math.Log2(float64(ivar.Type.Align())))
+		enc, err := encodeType(ivar.Type, false)
 		if err != nil {
-			return 0, fmt.Errorf("objc: couldn't add Ivar %s: %w", f.Name, err)
+			return 0, fmt.Errorf("objc: couldn't add Ivar %s: %w", ivar.Name, err)
 		}
-		if !class_addIvar(class, f.Name, size, alignment, enc) {
-			return 0, fmt.Errorf("objc: couldn't add Ivar %s", f.Name)
+		if !class_addIvar(class, ivar.Name, size, alignment, enc) {
+			return 0, fmt.Errorf("objc: couldn't add Ivar %s", ivar.Name)
 		}
-		if offset := class.InstanceVariable(f.Name).Offset(); offset != f.Offset {
-			return 0, fmt.Errorf("objc: couldn't add Ivar %s because offset (%d != %d)", f.Name, offset, f.Offset)
+		switch ivar.Attribute {
+		case ReadWrite:
+			ty := reflect.FuncOf(
+				[]reflect.Type{
+					reflect.TypeOf(ID(0)), reflect.TypeOf(SEL(0)), ivar.Type,
+				},
+				nil, false,
+			)
+			var encoding string
+			if encoding, err = encodeFunc(reflect.New(ty).Elem().Interface()); err != nil {
+				return 0, fmt.Errorf("objc: failed to create read method for '%s': %w", ivar.Name, err)
+			}
+			instanceVariable := class.InstanceVariable(ivar.Name)
+			val := reflect.MakeFunc(ty, func(args []reflect.Value) (results []reflect.Value) {
+				// on entry the first and second arguments are ID and SEL followed by the value
+				if len(args) != 3 {
+					panic(fmt.Errorf("objc: incorrect number of arguments"))
+				}
+				// Grab the pointer from using the Ivar and dereference it
+				var value ID
+				switch args[2].Kind() {
+				case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+					value = ID(args[2].Int())
+				case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+					value = ID(args[2].Uint())
+				case reflect.Float32, reflect.Float64:
+					value = ID(math.Float64bits(args[2].Float()))
+				case reflect.Bool:
+					if args[2].Bool() {
+						value = 1
+					}
+				case reflect.Pointer, reflect.UnsafePointer:
+					value = ID(args[2].Pointer())
+				default:
+					panic(fmt.Errorf("objc: unsupported kind %s", args[2].Kind()))
+				}
+				object_setIvar(args[0].Interface().(ID), instanceVariable, value)
+				return nil
+			}).Interface()
+			class.AddMethod(RegisterName("set"+strings.Title(ivar.Name)+":\x00"), NewIMP(val), encoding)
```

Comment:
`strings.Title` is deprecated. https://pkg.go.dev/strings#Title . Also, we don't have to consider the case when `name` includes space chars.

What about making our own function to make a string a setter name?

```go
func toSetterName(name string) string {
    // This assumes name consists of ASCII characters.
    // I don't think we have to consider non-ASCII characters.
    return "set" + strings.ToUpper(string(name[0])) + name[1:]
}
```

Reply:
Also, for a boolean member, `is` should be used as a prefix.